cogs5e/models/automation/utils.py

```diff
@@ -64,13 +64,38 @@ def max_mapper(node: d20.ast.Node):
     return node
 
 
+def max_add_crit_mapper(node: d20.ast.Node):
+    """A function that adds the maximum value of each Dice AST node as a literal"""
+    if isinstance(node, d20.ast.Dice):
+        return d20.ast.BinOp(node, "+", d20.ast.Literal(node.num * node.size))
+    return node
+
+
 def crit_mapper(node: d20.ast.Node):
     """A function that doubles the number of dice for each Dice AST node."""
     if isinstance(node, d20.ast.Dice):
         return d20.ast.Dice(node.num * 2, node.size)
     return node
 
 
+def double_dice_crit_mapper(node: d20.ast.Node):
+    """A function that doubles the number of dice for each Dice AST node."""
+    if isinstance(node, d20.ast.Dice):
+        return d20.ast.BinOp(d20.ast.Parenthetical(node), "*", d20.ast.Literal(2))
+    return node
+
+
+def crit_dice_gen(dice_ast: d20.ast.Node, critdice: int):
+    """A function that finds the size of left most Dice AST node and generates crit dice based on that."""
```

Comment:
I would make this docstring more detailed - specifically, to answer the question "how does it generate crit dice given the size of the leftmost dice ast node?" without having to read the source

*insert based dice joke here*

Reply:
This code might be unnecessary, see comment above